modules/cloudsql-postgres/variables.tf

```diff
@@ -0,0 +1,151 @@
+variable "name" {
+  description = "Cloud SQL instance name (lowercase letters, numbers, and hyphens; up to 98 characters)."
+  type        = string
+}
+
+variable "project" {
+  description = "GCP project ID hosting the Cloud SQL instance."
+  type        = string
+}
+
+variable "region" {
+  description = "GCP region for the primary instance."
+  type        = string
+}
+
+variable "network" {
+  description = "Self‑link or name of the VPC network used for private IP connectivity."
+  type        = string
+}
+
+variable "require_squad" {
+  description = "Whether the `squad` variable must be specified."
+  type        = bool
+  default     = true
+}
+
+variable "squad" {
+  description = "Squad or team label applied to the instance."
+  type        = string
+  default     = ""
+
+  validation {
+    condition     = !var.require_squad || var.squad != ""
+    error_message = "squad must be specified or disable the check by setting require_squad = false."
+  }
+}
+
+# Engine & Capacity
+
+variable "database_version" {
+  description = "Cloud SQL engine version."
+  type        = string
+  default     = "POSTGRES_17"
+}
```

Comment:
we should add a validation to this variable that the database_version is prefixed with `POSTGRES_` to ensure users dont accidentally run mysql/sql server with this module

https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules